performance_tests/README.rst

```diff
@@ -0,0 +1,232 @@
+                                    if n_iters = 100, this performance
+                                    test script will run the create_keyring
+                                    method 100 times and report the
+                                    execution time of each of the calls.
+      --output_file OUTPUT_FILE     The output file for execution times
+                                    for each function call,
+                                    default='kms_keyring_create' in the
+                                    results folder.
+
+
+Consolidate Results
+~~~~~~~~~~~~~~~~~~~
+
+In order to find the minimum, maximum, average, 99th percentile and bottom
+99th percentile trimmed average times from the n_iters runs, please use the
+following script from the performance_tests directory:
+
+.. code::
+
+    usage: python consolidate_results.py results/kms_keyring_create.csv
+
+
+Encrypt
+-------
+
+To run the performance test for execution time, please use the following
+commands in the performance_tests directory:
+
+.. code::
+
+    usage: python test/keyrings/test_aws_kms_keyring.py encrypt
+
+    optional arguments:
+      -h, --help                                            show this help message and exit.
+      --plaintext_data_filename PLAINTEXT_DATA_FILENAME     Filename containing plaintext data
+                                                            you want to encrypt.
+                                                            default='test/resources/plaintext/plaintext-data-medium.dat'.
+                                                            You can choose to use any other plaintext
+                                                            file as well. Some example plaintext
+                                                            data files are present in the
+                                                            'test/resources' directory.
+      --kms_key_id KMS_KEY_ID                               The KMS key ID you want to use.
+      --n_iters N_ITERS                                     Number of iterations you want to
+                                                            run the test for. For instance,
+                                                            if n_iters = 100, this performance
+                                                            test script will run the create_keyring
+                                                            method 100 times and report the
+                                                            execution time of each of the calls.
+      --output_file OUTPUT_FILE                             The output file for execution times
+                                                            for each function call,
+                                                            default='kms_keyring_create' in the
+                                                            results folder.
+
+Consolidate Results
```

Comment:
Can the 3 "Consolidate Results" sections be... consolidated? (ba dum tss)
Since it's all just the same invocation with a different filename

Reply:
Haha, yeah done.